### ISSUE TITLE
http_archive download_and_extract do not allow with non-ASCII characters in file names

### ISSUE THREAD

**Issue by mikhail-g** [bazelbuild/bazel#18254]
### Description of the bug:

Function `download_and_extract` from rule `http_archive` Failed to extract an archive with a file containing a non-ASCII character in its name `Āfoo.go`

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Example BUILD.bazel:
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "go_1_16",
    urls = ["https://dl.google.com/go/go1.16.7.linux-amd64.tar.gz"]
    sha256 = "7fe7a73f55ba3e2285da36f8b085e5c0159e9564ef5f63ee0ed6b818ade8ef04",
    build_file = "//build_tools/go:BUILD.go-dist",
)
```

### Which operating system are you running Bazel on?

MacOS Monterey

### What is the output of `bazel info release`?

release 4.2.2

### Any other information, logs, or outputs that you want to share?

Stacktrace:
```
Repository rule http_archive defined at:
repo/http.bzl:336:31: in <toplevel>
ERROR: An error occurred during the fetch of repository 'go_1_16':
   Traceback (most recent call last):
	File ".../external/bazel_tools/tools/build_defs/repo/http.bzl", line 111, column 45, in _http_archive_impl
		download_info = ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error extracting .../go_1_16/temp3086185247349608877/go1.16.7.linux-amd64.tar.gz to .../go_1_16/temp3086185247349608877: .../go_1_16/go/test/fixedbugs/issue27836.dir/?foo.go (Illegal byte sequence)
```

**Comment by meteorcloudy** [comment]
Thanks for reporting! Is this still an issue in the latest release of Bazel? A PR for fixing this is welcome!

**Comment by fmeum** [comment]
This has been fixed by 10169bbe66e818868ec37f7d853ab5a567cd2ced, which made it into Bazel 7 and 6.4.0.
